src/ts/vector.ts

```diff
@@ -17,3 +17,7 @@ export function vecAngleBetween(a: Vector, b: Vector) {
 export function vecFromAngleAndScale(radians: number, scale = 1) {
   return { x: Math.cos(radians) * scale, y: Math.sin(radians) * scale };
 }
+
+export function vecEquals(a: Vector, b: Vector): boolean {
+  return a.x === b.x && a.y === b.y && a.dx === a.dx && a.dy === b.dy;
```

Comment:
We wouldn't have one vector with all 4 of these properties would we? I think we should change velocity to use x and y, there's no reason to use different properties

Reply:
I also am not sure why we have a vector with `dx` and `dy` but given that we do, this followed the principle of least surprise to check all of them. I'm fully in favor of splitting velocity away from position and having just one vector.